uswds/uswds
# USWDS - Pagination: Remove presentation role

# Summary

This PR removes the role attribute from our pagination component. It also updates `axe-core` and `axe-storybook-testing` dev dependencies.

## Breaking change

:warning: This is a breaking change.


## Related issue

Closes #5022 

## Preview link
[Pagination →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-pagination-remove-role/iframe.html?args=&id=components-pagination--default&viewMode=story)

Preview link:


## Problem statement

WCAG 2.1 & 2.0 guidelines require `ul` & `ol` to contain only `li` **content** elements. Elements containing `role=presentation` are not allowed under this guideline.

## Solution
Remove role from pagination component.

## Testing and review

1. Use [pagination component](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-pagination-remove-role/iframe.html?args=&id=components-pagination--default&viewMode=story) with screen reader
2. There should be no change in readouts from its [develop counterpart](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/iframe.html?id=components-pagination--default&viewMode=story)
3. Run `npm test:a11y` and ensure no failing tests
4. Use axe-core browser plugin on component page to ensure no failing tests

## Dependency updates

| Dependency name              | Previous version | New version |
| ---------------------------- | :--------------: | :---------: |
| axe-core |     [4.3.4]      |   [^4.6.3]   |
| [@chanzuckerberg/axe-storybook-testing]     |        [5.0.1]        |   [^6.3.0]   |

Before opening this PR, make sure you’ve done whichever of these applies to you:
